Test reporter fails to build under 2.12.6

Attempting to run a test under 2.12.6 appears to fail due to `-Xfatal-warnings` + unused pattern vars. I'm not sure why this wasn't caught with CI.
```
ERROR: /home/sam/.cache/bazel/_bazel_sam/d50f718107f06eb5b720bb473093cd22/external/io_bazel_rules_scala/scala/support/BUILD:3:1: scala @io_bazel_rules_scala//scala/support:test_reporter failed (Exit 1)
external/io_bazel_rules_scala/scala/support/JUnitXmlReporter.scala:62: warning: pattern var e in method apply is never used; `e@_' suppresses this warning
      case e: RunCompleted =>
           ^
...
external/io_bazel_rules_scala/scala/support/JUnitXmlReporter.scala:458: warning: pattern var e in lazy value hostname is never used; `e@_' suppresses this warning
      case e: UnknownHostException => "unknown"
           ^
error: No warnings can be incurred under -Xfatal-warnings.
19 warnings found
one error found
19 warnings found
one error found
java.lang.RuntimeException: Build failed
        at io.bazel.rulesscala.scalac.ScalacProcessor.compileScalaSources(ScalacProcessor.java:242)
        at io.bazel.rulesscala.scalac.ScalacProcessor.processRequest(ScalacProcessor.java:67)
        at io.bazel.rulesscala.worker.GenericWorker.runPersistentWorker(GenericWorker.java:45)
        at io.bazel.rulesscala.worker.GenericWorker.run(GenericWorker.java:111)
        at io.bazel.rulesscala.scalac.ScalaCInvoker.main(ScalaCInvoker.java:41)
```

Will follow up shortly with a PR that does as suggested, but I'm very curious as to why this isn't caught in CI

--- a/scala/support/JUnitXmlReporter.scala
+++ b/scala/support/JUnitXmlReporter.scala
@@ -59,13 +59,13 @@ class JUnitXmlReporter extends Reporter {
       case e: SuiteAborted =>
         testSuites += getTestsuite(e, e.suiteId)
 
-      case e: RunCompleted =>
+      case _: RunCompleted =>
         write(testSuites)
         testSuites = Set.empty
-      case e: RunStopped =>
+      case _: RunStopped =>
         write(testSuites)
         testSuites = Set.empty
-      case e: RunAborted =>
+      case _: RunAborted =>
         write(testSuites)
         testSuites = Set.empty
 
@@ -193,13 +193,13 @@ class JUnitXmlReporter extends Reporter {
           testsuite.testcases += testcase
           idx += 1
 
-        case e: InfoProvided   => idx += 1
-        case e: AlertProvided  => idx += 1
-        case e: NoteProvided   => idx += 1
-        case e: MarkupProvided => idx += 1
-        case e: ScopeOpened    => idx += 1
-        case e: ScopeClosed    => idx += 1
-        case e: ScopePending   => idx += 1
+        case _: InfoProvided   => idx += 1
+        case _: AlertProvided  => idx += 1
+        case _: NoteProvided   => idx += 1
+        case _: MarkupProvided => idx += 1
+        case _: ScopeOpened    => idx += 1
+        case _: ScopeClosed    => idx += 1
+        case _: ScopePending   => idx += 1
         case e: TestPending    => unexpected(e)
         case e: TestCanceled   => unexpected(e)
         case e: RunStarting    => unexpected(e)
@@ -246,7 +246,7 @@ class JUnitXmlReporter extends Reporter {
       val event = orderedEvents(idx)
 
       event match {
-        case e: SuiteStarting =>
+        case _: SuiteStarting =>
           startIndex = idx
 
         case e: SuiteCompleted =>
@@ -322,13 +322,13 @@ class JUnitXmlReporter extends Reporter {
           testcase.canceled = true
           idx += idxAdjustmentForRecordedEvents(e.recordedEvents)
 
-        case e: ScopeOpened    => idx += 1
-        case e: ScopeClosed    => idx += 1
-        case e: ScopePending   => idx += 1
-        case e: InfoProvided   => idx += 1
-        case e: MarkupProvided => idx += 1
-        case e: AlertProvided  => idx += 1
-        case e: NoteProvided   => idx += 1
+        case _: ScopeOpened    => idx += 1
+        case _: ScopeClosed    => idx += 1
+        case _: ScopePending   => idx += 1
+        case _: InfoProvided   => idx += 1
+        case _: MarkupProvided => idx += 1
+        case _: AlertProvided  => idx += 1
+        case _: NoteProvided   => idx += 1
         case e: SuiteCompleted => unexpected(e)
         case e: TestStarting   => unexpected(e)
         case e: TestIgnored    => unexpected(e)
@@ -455,7 +455,7 @@ class JUnitXmlReporter extends Reporter {
       val localMachine = InetAddress.getLocalHost();
       localMachine.getHostName
     } catch {
-      case e: UnknownHostException => "unknown"
+      case _: UnknownHostException => "unknown"
     }
 
   //

